sys/dev/nvme/nvme_qpair.c

```diff
@@ -1094,15 +1094,52 @@ nvme_payload_map(void *arg, bus_dma_segment_t *seg, int nseg, int error)
 	 * Note that we specified ctrlr->page_size for alignment and max
 	 * segment size when creating the bus dma tags.  So here we can safely
 	 * just transfer each segment to its associated PRP entry.
+	 * 
+	 * According to NVMe specification, PRP entries must be 4-byte aligned.
+	 * Validate and ensure all PRP addresses are properly aligned.
 	 */
+	
+	/* Validate PRP1 alignment - must be 4-byte aligned */
+	if ((seg[0].ds_addr & 3) != 0) {
```

Comment:
The transfer address should have been validated before we get to here. This is the wrong place to check.